### ISSUE TITLE
Time series query broken 

### ISSUE THREAD

**Issue by jzhao62** [apache/superset#27798]
### Bug description

- the query used to generate time series data some times needs be explicitly casted as timestamp, in order to be used in Postgres

### How to reproduce the bug

1. dump attached data to a a connected Postgres 
2. mark the timestamp as Temporal data
3. create any time based chart and examine the auto generated sql 
[time-series-data.csv](https://github.com/apache/superset/files/14820362/time-series-data.csv)

### Screenshots/recordings

![Screenshot_20240331_233316](https://github.com/apache/superset/assets/25937657/1108932f-3ca8-4202-958e-ae205ccba2a7)
![Screenshot_20240331_233212](https://github.com/apache/superset/assets/25937657/37776492-7312-4ede-b7ec-f1d8d8186a52)
![Screenshot_20240331_233241](https://github.com/apache/superset/assets/25937657/44b68c90-4c30-4bbd-935b-5e19b40df7df)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by zhaoyongjie** [comment]
Hey @jzhao62, there is no column typing convert logic in current time grain expression so the TIMESTAMP column should have to be offered. The quick and dirty workaround to create a` computed column` in Superset dataset and apply the `computed column` as time column as your use case.

**Comment by jzhao62** [comment]
hey @zhaoyongjie thanks, i dont know if this situation applies to all time series data or not. If we explicilty cast the timestamp when generating sql on UI, would it be a problem ?

**Comment by zhaoyongjie** [comment]
@jzhao62 I totally agree with you. Ideally, the explicit typing conversion should be applied on the temporal column if explicitly defined a column as datetime/date/timestamp.

**Comment by rusackas** [comment]
This bug is reported on an unsupported superset version, and has gone quite stale. I'll close it as largely inactionable, since it sounds like we need a design/volunteer for a feature that handles these timestamp conversions automatically, or this is just something users have to do, and it's a query pain moreso than a bug. If I'm misconstruing, and/or current versions still have an actual bug, let me know and we can reopen this.
